### ISSUE TITLE
Windows triggerRestart() not working

### ISSUE THREAD

**Issue by chicknsoup** [jpillora/overseer#51]
Similar to watchParrent(), we cannot send signals between windows processes, so triggerRestart() obviously does not work.

To fix this, we have to find a way to send/receive messages between master/slave processes.

* gRPC over tcp
* RESTful
* Windows API (like postmessage/sendmessage)

What do you think? @jpillora

**Comment by jpillora** [comment]
It would be good to swap from signals to a more cross-platform approach.
Can windows use net.ListenUnix? If so I'm thinking we listen with a random
socket file, then we pass the filename into the child process. Then we can
have proper bi-directional communications, where we remove all of the
signal handling (except user restarts) and use socket file tcp instead

**Comment by chicknsoup** [comment]
Yes, recent versions of Windows support AF_UNIX, so `net.ListenUnix` works. Just did a quick test  on my Windows 10 machine, it works as expected.
